packages/core/src/exporter/download-videos.ts

```diff
@@ -0,0 +1,58 @@
+import {AssetInfo} from '../app';
+
+export async function download(assets: AssetInfo[][]): Promise<void> {
+  const videoRanges: Map<string, {start: number; end: number}> = new Map();
+
+  assets.forEach(frameAssets => {
```

Comment:
When ended up with nested if/elses like this please consider rewriting it to make it more readable.
```ts
  assets.forEach(frameAssets => {
    frameAssets.forEach(asset => {
      if (asset.type !== 'video' || asset.decoder !== 'ffmpeg') {
        return;
      }

      if (videoRanges.has(asset.src)) {
          const range = videoRanges.get(asset.src)!;
          range.start = Math.min(range.start, asset.currentTime);
          range.end = Math.max(range.end, asset.currentTime);
          return;
      }

      videoRanges.set(asset.src, {
        start: asset.currentTime,
        end: asset.currentTime,
      });
    });
  });
  ```